lib/compiler-singlepass/src/compiler.rs

```diff
@@ -113,6 +115,20 @@ impl Compiler for SinglepassCompiler {
                 tracing::info_span!("function", i = i.index()).in_scope(|| {
                     let reader =
                         wasmer_compiler::FunctionReader::new(input.module_offset, input.data);
+                    let stack_init_gas_cost = instrumentation.function_frame_sizes[i.index()]
+                        .checked_mul(tunables.regular_op_cost() / 8) // can run i64.const, so can init 8 bytes within one regular op
```

Comment:
Probably should make this a separate tunable too, rather than hardcoding a specific factor here. e.g. if `regular_op_cost` were `< 8` this would also charge 0 to initialize locals overall, every time, having a separate tunable gives the embedder (nearcore) a decision to make if they want to keep `regular_op_cost < 8` or do something else.

Reply:
I was thinking this should happen at the same time as we’ll be doing per-instruction gas weights. In the meantime, I don’t think we would ever consider having a regular op cost less than 8 (or for which dividing by 8 would round too badly), seeing how big our current numbers are